src/services/utilities.ts

```diff
@@ -1363,6 +1363,10 @@ namespace ts {
         return textSpanContainsPosition(span, node.getStart(file)) &&
             node.getEnd() <= textSpanEnd(span);
     }
+
+    export function getTypesOfUnion(type: Type): ReadonlyArray<Type> {

```

Comment:
I think we need better name as the name suggests that it handles only union.

Reply:
Also this creates unnecessary array for the non union type. Depending on how frequently this would be called, that's too many arrays for no reason?